Menu adjustments

Allcharles opened the pull request:
# Adjustments to Menu System

Minor adjustments to menu system to remove small bugs.

## Changes

- Set links with no order to a value of `Infinity` so they appear on the bottom of the list
- Updated `MenuLink` items to accept a `param` value from the router. This allows links to direct to the ecosounds website
- Updated secondary menu links so that current breadcrumbs are visible regardless of predicate
- Minor changes to unit tests

## Closes

Closes #98 

## Visual Changes

Breadcrumbs appear in the below image, even though the user is not logged in and shouldn't be able to view the final link.

![image](https://user-images.githubusercontent.com/3955116/75007476-035bf180-54c1-11ea-9232-df097baeccbd.png)

Allcharles committed:
Fixed un-ordered link ordering

Allcharles committed:
Updated Href to pass baw url and router params

Allcharles committed:
Fixed unit tests

Removed bawUrl from MenuLink

Allcharles committed:
Fixed missing self links

atruskie commented on src/app/component/projects/projects.menus.ts:
```diff
@@ -72,7 +72,7 @@ export const editProjectMenuItem = MenuRoute({
 
 export const exploreAudioProjectMenuItem = MenuLink({
-  uri: "REPLACE_ME",
+  uri: params => `/visualize?projectId=${params.projectId}`,
```
use our string templating function that was made for api services... 

Many different pages may want to create a visualize link.

atruskie commented on src/app/component/shared/header/header-dropdown/header-dropdown.component.ts:
```diff
@@ -54,4 +55,16 @@ export class HeaderDropdownComponent implements OnInit {
   constructor() {}
 
   ngOnInit() {}
+
```
this looks like it is reused. Move it somewhere appropriate and remove copies in other files

atruskie reviewed (approved):
Minor changes, no further approval needed.

Allcharles committed:
Extracted getRoute function

Resolves: https://github.com/QutEcoacoustics/workbench-client/pull/109#discussion_r381746458

Allcharles committed:
Cleaned up menu items

Allcharles committed:
Alternative method to bypass predicate

Allcharles committed:
Added check for duplicate self links
